python/cuml/linear_model/logistic_regression.pyx

```diff
@@ -314,6 +337,38 @@ class LogisticRegression(Base):
         """
         return self.qn.predict(X, convert_dtype=convert_dtype)
 
+    def predict_proba(self, X, convert_dtype=False):
+        """
+        Predicts the class probabilities for X
+
+        Parameters
+        ----------
+        X : array-like (device or host) shape = (n_samples, n_features)
+            Dense matrix (floats or doubles) of shape (n_samples, n_features).
+            Acceptable formats: cuDF DataFrame, NumPy ndarray, Numba device
+            ndarray, cuda array interface compliant array like CuPy
+
+        convert_dtype : bool, optional (default = False)
+            When set to True, the predict method will, when necessary, convert
+            the input to the data type which was used to train the model. This
+            will increase memory used for the method.
+        Returns
+        ----------
+        y: cuDF DataFrame
+           Dense matrix (floats or doubles) of shape (n_samples, C)
+        """
+        proba = self.decision_function(X, convert_dtype=convert_dtype)
+        proba = cp.asarray(proba.to_gpu_matrix())
+        if self.num_classes == 2:
+            proba = 1 / (1 + cp.exp(-proba))
+            proba = cp.column_stack((proba, proba))
+            proba[:, 0] = 1 - proba[:, 0]
+        elif self.num_classes > 2:
+            row_sum = cp.sum(proba, axis=1)
```

Comment:
reading this code, it looks like in the 2-class case, decision_function is returning the pre-sigmoid scores, but maybe already-exp-transformed scores in the softmax case? Maybe I'm confused... just seems like we should have an exp transform in the multiclass case here or no logit transform in the binary case

Reply:
Nope, I discussed this w/ him as well. The exponentials need to be added to the softmax case.

Reply:
you guys are always ahead of me! should be an easy fix. Looks like the change is not too far off from completion, mostly needing tests and these small tweaks.